pkg/intermediate/aggregate_test.go

```diff
@@ -694,6 +694,66 @@ func TestGetExpiryFromExpirePriorityQueue(t *testing.T) {
 	}
 }
 
+func TestGetRecords(t *testing.T) {
+	messageChan := make(chan *entities.Message)
+	input := AggregationInput{
+		MessageChan:           messageChan,
+		WorkerNum:             2,
+		CorrelateFields:       fields,
+		ActiveExpiryTimeout:   testActiveExpiry,
+		InactiveExpiryTimeout: testInactiveExpiry,
+	}
+	ap, _ := InitAggregationProcess(input)
+
+	// Add records with IPv4 fields.
+	recordIPv4Src := createDataMsgForSrc(t, false, false, false, false, false).GetSet().GetRecords()[0]
+	recordIPv4Dst := createDataMsgForDst(t, false, false, false, false, false).GetSet().GetRecords()[0]
+	// Add records with IPv6 fields.
+	recordIPv6Src := createDataMsgForSrc(t, true, false, false, false, false).GetSet().GetRecords()[0]
+	recordIPv6Dst := createDataMsgForDst(t, true, false, false, false, false).GetSet().GetRecords()[0]
+
+	records := []entities.Record{recordIPv4Src, recordIPv4Dst, recordIPv6Src, recordIPv6Dst}
+
+	flowKeyIPv4, _, _ := getFlowKeyFromRecord(recordIPv4Src)
+
+	for _, record := range records {
+		flowKey, isIPv4, _ := getFlowKeyFromRecord(record)
+		err := ap.addOrUpdateRecordInMap(flowKey, record, isIPv4)
+		assert.NoError(t, err)
+	}
+	recordStrings := ap.GetRecords(flowKeyIPv4)
+	assert.Equalf(t, 1, len(recordStrings), "Number of IPv4 records string is incorrect, expected 1 got %d", len(recordStrings))
+	assert.Contains(t, recordStrings[0], "    sourceTransportPort: 1234 \n")
+	assert.Contains(t, recordStrings[0], "    destinationTransportPort: 5678 \n")
+	assert.Contains(t, recordStrings[0], "    protocolIdentifier: 6 \n")
+	assert.Contains(t, recordStrings[0], "    sourcePodName: pod1 \n")
+	assert.Contains(t, recordStrings[0], "    destinationPodName: pod2 \n")
+	assert.Contains(t, recordStrings[0], "    sourceIPv4Address: 10.0.0.1 \n")
+	assert.Contains(t, recordStrings[0], "    destinationIPv4Address: 10.0.0.2 \n")
+	assert.Contains(t, recordStrings[0], "    destinationClusterIPv4: 192.168.0.1 \n")
+	assert.Contains(t, recordStrings[0], "    destinationServicePort: 4739 \n")
+	assert.Contains(t, recordStrings[0], "    flowEndSeconds: 1 \n")
+	assert.Contains(t, recordStrings[0], "    flowType: 2 \n")
+	assert.Contains(t, recordStrings[0], "    flowEndReason: 2 \n")
+	assert.Contains(t, recordStrings[0], "    tcpState: ESTABLISHED \n")
+	assert.Contains(t, recordStrings[0], "    ingressNetworkPolicyRuleAction: 0 \n")
+	assert.Contains(t, recordStrings[0], "    egressNetworkPolicyRuleAction: 0 \n")
+	assert.Contains(t, recordStrings[0], "    ingressNetworkPolicyRulePriority: 50000 \n")
+	assert.Contains(t, recordStrings[0], "    packetTotalCount: 500 \n")
+	assert.Contains(t, recordStrings[0], "    packetDeltaCount: 0 \n")
+	assert.Contains(t, recordStrings[0], "    reversePacketTotalCount: 500 \n")
+	assert.Contains(t, recordStrings[0], "    reversePacketDeltaCount: 0 \n")
+
+	partialFlowKeyIPv6 := &FlowKey{
+		SourceAddress: "2001:0:3238:dfe1:63::fefb",
+	}
+	recordStrings = ap.GetRecords(partialFlowKeyIPv6)
```

Comment:
Understand there are lots of same fields for v4 and v6 records so you omitted some assert checks. 
Could we have a seperate function containing these asserts with ip version as an argument to reuse asserts?

Reply:
Sounds good. Updated